### ISSUE TITLE
Crash when building StateManager

### ISSUE THREAD

**Issue by MatteoNasci** [jiangzhongshi/Scaffold-Map#12]
`StateManager s_(DemoType::PARAM, "D:/camel_b.obj");`
[camel_b.zip](https://github.com/jiangzhongshi/Scaffold-Map/files/4091101/camel_b.zip)

When I attempt to run my code it crashes at this line. 
Before crashing there is 'Mesh Measure7854.29' in output so I believe the execution reaches at least StateManager.cpp line 90 since that debug comes from the add_new_patch method (ScafData.cpp line 394). 
What may be the problem?

**Comment by jiangzhongshi** [comment]
Sorry, I am swamped by a deadline now, I can check next week. Is this a
mesh of mine?

On Tue, Jan 21, 2020 at 7:51 AM Matteo Lorenzo Nasci <
notifications@github.com> wrote:

> StateManager s_(DemoType::PARAM, "D:/camel_b.obj");
> camel_b.zip
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jiangzhongshi_Scaffold-2DMap_files_4091101_camel-5Fb.zip&d=DwMCaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=2v3jiwtgToNyGgWlUiot8g&m=Ar5MA73JR9qg3BUl3buw2MnXhqkEN-iNgiLg6TIvkek&s=nBo4PLkumbHJuCpKFk0OkAkoCwASG3Bc3Sjpd4RL_Ac&e=>
>
> When I attempt to run my code it crashes at this line.
> Before crashing there is 'Mesh Measure7854.29' in output so I believe the
> execution reaches at least StateManager.cpp line 90 since that debug comes
> from the add_new_patch method (ScafData.cpp line 394).
> What may be the problem?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jiangzhongshi_Scaffold-2DMap_issues_12-3Femail-5Fsource-3Dnotifications-26email-5Ftoken-3DADBLCF3CVDAE7PYEOIFDFU3Q63VW3A5CNFSM4KJS5QY2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4IHT6FEA&d=DwMCaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=2v3jiwtgToNyGgWlUiot8g&m=Ar5MA73JR9qg3BUl3buw2MnXhqkEN-iNgiLg6TIvkek&s=hZpSpB7eAaC_8huMK14OdXyO1PjiXFWGuMPRw1Ovaig&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ADBLCFYRJWY3BHU4ZR562ADQ63VW3ANCNFSM4KJS5QYQ&d=DwMCaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=2v3jiwtgToNyGgWlUiot8g&m=Ar5MA73JR9qg3BUl3buw2MnXhqkEN-iNgiLg6TIvkek&s=FPH8UsTYSBr-U6US5FUdcwE29zv5C5s4eNJMmafVM14&e=>
> .
>


-- 
Best,

Zhongshi Jiang
New York University

**Comment by MatteoNasci** [comment]
Yes, it's the mesh included in the repository (I am using it to test).

**Comment by jiangzhongshi** [comment]
Hi,

Could you show me the complete main file so that I can try to reproduce the error?

Thanks
